src/wallet/db.cpp

```diff
@@ -672,6 +672,12 @@ bool CWalletDBWrapper::Backup(const std::string& strDest)
                     pathDest /= strFile;
 
                 try {
+                    if (fs::equivalent(pathSrc, pathDest)) {
+                        LogPrintf("cannot backup to wallet source file %s\n",
+                            pathDest.string());
```

Comment:
nit: the repository doesn't enforce max line length, so you can join this with the line above.